### ISSUE TITLE
TimeoutError in Client.join_channels causes unhandled exception

### ISSUE THREAD

**Issue by SamChill** [TwitchIO/TwitchIO#209]
If `Client.join_channels` is called and a `TimeoutError` occurs while trying to join a channel, then an unhandled exception will be raised when this channel is removed from the `_initial_channels` list, because the channel is not an initial channel, but one passed in via `join_channels`.
```
Traceback (most recent call last):
  File "/home/chill/.cache/pypoetry/virtualenvs/timedoutbot-5LVMUk6d-py3.7/lib/python3.7/site-packages/twitchio/websocket.py", line 265, in _join_future_handle
    await asyncio.wait_for(fut, timeout=10)
  File "/home/chill/.pyenv/versions/3.7.1/lib/python3.7/asyncio/tasks.py", line 423, in wait_for
    raise futures.TimeoutError()
concurrent.futures._base.TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/chill/.cache/pypoetry/virtualenvs/timedoutbot-5LVMUk6d-py3.7/lib/python3.7/site-packages/twitchio/websocket.py", line 275, in _join_future_handle
    await self._process_data(data)
  File "/home/chill/.cache/pypoetry/virtualenvs/timedoutbot-5LVMUk6d-py3.7/lib/python3.7/site-packages/twitchio/websocket.py", line 284, in _process_data
    return await self._code(parsed, parsed["code"])
  File "/home/chill/.cache/pypoetry/virtualenvs/timedoutbot-5LVMUk6d-py3.7/lib/python3.7/site-packages/twitchio/websocket.py", line 321, in _code
    self._initial_channels.remove(parsed["batches"][0])
ValueError: list.remove(x): x not in list
```

**Comment by Commaster** [comment]
When addressing this is it possible to distinguish between actual (network) timeout and a chat timeout/ban?
And possibly event create new `event_`s for each individual outcome?

**Comment by chillymosh** [comment]
This may have already been resolved in one of the recent versions or master.
Network timeouts to the websocket are a totally different thing and have their own errors raised.

**Comment by Commaster** [comment]
Still not fixed in 2.4.0: (Either renamed/suspended channel or chat ban)
```
The channel "%REDACTED%" was unable to be joined. Check the channel is valid.
```
Followed immediately by:
```
Task exception was never retrieved
future: <Task finished name='Task-13906' coro=<WSConnection._join_future_handle() done, defined at .../python3.8/site-packages/twitchio/websocket.py:293> exception=ValueError('list.remove(x): x not in list')>
Traceback (most recent call last):
  File ".../python3.8/site-packages/twitchio/websocket.py", line 295, in _join_future_handle
    await asyncio.wait_for(fut, timeout=timeout)
  File ".../python3.8/asyncio/tasks.py", line 501, in wait_for
    raise exceptions.TimeoutError()
asyncio.exceptions.TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".../python3.8/site-packages/twitchio/websocket.py", line 305, in _join_future_handle
    await self._process_data(data)
  File ".../python3.8/site-packages/twitchio/websocket.py", line 314, in _process_data
    return await self._code(parsed, parsed["code"])
  File ".../python3.8/site-packages/twitchio/websocket.py", line 347, in _code
    self._initial_channels.remove(parsed["batches"][0])
ValueError: list.remove(x): x not in list
```

P.S. And 🙏 please add `event_join_failed` or similar to deal with such situations automatically instead of reading the log file *eventually*. Thank you!

**Comment by chillymosh** [comment]
How many channels are you joining in your `initial_channels`? Are you joining any channels manually after using `join_channels()` if so then how many channels there in a single call and how long after the bot has hit event_ready are you calling it?

**Comment by Commaster** [comment]
1. 0, the list is empty (see #290).
2. Yes, one at a time. In this case it has been at least 30 minutes after `event_ready`. (Restarting for 2.4.0 update)
3. No. But I get no issues joining other valid/existing channels. This only happens if you try to join a suspended/renamed/invalid channel or the chat is in subscriber-only mode or the bot account is banned from chat.
4. I will see if this can be arranged, just using what the distro provides 😅 

tl;dr: The first message `Check the channel is valid.` is correct, the channel is not valid or you are not allowed to join chat. The issue is in handling the 353 code from https://github.com/TwitchIO/TwitchIO/blob/master/twitchio/websocket.py#L301

**Comment by chillymosh** [comment]
Event_ready now fires in 2.4 with empty initial_channels, as per the changelog, although this issue is isolated to calling join_channels manually.

**Comment by Commaster** [comment]
I'm still wondering if it's possible to differentiate chat bans from follower-only and/or subscriber-only chat (if the bot account is not a follower/subscriber).
https://dev.twitch.tv/docs/irc/tags#roomstate-tags
Not sure if we receive these tags when not allowed to join the room/chat...
> The Twitch IRC server sends this message ***after*** a bot joins a channel

**Comment by chillymosh** [comment]
If you are banned from a channel you receive
```
DEBUG:twitchio.websocket: < @msg-id=msg_banned :tmi.twitch.tv NOTICE #channel :You are permanently banned from talking in channel.
```

**Comment by Commaster** [comment]
Is this handled and exposed by TwitchIO somehow? Or everything falls to "unable to be joined"?
